### PR TITLE
Add properties about header/body validation split 

### LINKED ISSUE #535: Add time-traveling block header validation property for Byron
State the property that states that we can travel back in time `k` blocks and forward in time `2k` slots and validate headers using a subset of the ledger state.

A visual sketch of this property is shown below:

image

Stating the property for some abstract function can be a first step. We can later make this function concrete by adapting the ledger and chain rules.

Related to https://github.com/input-output-hk/cardano-ledger-specs/issues/529

### PR DESCRIPTION
**Pull request by dnadales**
Closes #535

### CODE REVIEW

File: byron/chain/formal-spec/blockchain-spec.tex [hunk 1]
```diff
@@ -1300,9 +1305,198 @@ \section{Blockchain Extension}
...
+block headers without having to download the block bodies.
+Property~\ref{prop:header-only-validation} states that if an extension of a
+chain that spans less than $2 \cdot k$ slots is valid, then validating the
+headers of that extension is also valid. This property is useful for its
+converse: if the header validation check for a sequence of headers does not
+pass, then we know that the block validation that corresponds to those headers
+will not pass either.
+
+\begin{property}[Header only validation]\label{prop:header-only-validation}
+  For all environments $e$, states $s$ with slot number ($\var{s_{last}}$) $t$;
```
[the remark this answers is not in the document]
Comment (mdimjasevic) [reply byron/chain/formal-spec/blockchain-spec.tex]: Can you clarify what is "(slast ) t;" supposed to mean?

File: byron/chain/formal-spec/blockchain-spec.tex [hunk 2]
```diff
@@ -1300,9 +1305,198 @@ \section{Blockchain Extension}
 \caption{Blockchain extension rules}
 \label{fig:rules:chain-extension}
 \end{figure}
+
+\clearpage
+
+\section{Transition Systems Properties}
```
[the remark this answers is not in the document]
Comment (dnadales) [reply byron/chain/formal-spec/blockchain-spec.tex]: But this section contains the properties of the transition systems defined in the document. Hence the plural.

File: byron/chain/formal-spec/blockchain-spec.tex [hunk 3]
```diff
@@ -1300,9 +1305,198 @@ \section{Blockchain Extension}
...
+\label{sec:ts-properties}
+
+\subsection{Header only validation}
```
[the remark this answers is not in the document]
Comment (dnadales) [reply byron/chain/formal-spec/blockchain-spec.tex]: Actually I noticed that in the rest of the document we don't use title case, so I'll change the other Title Case I introduced in this PR.

File: byron/chain/formal-spec/blockchain-spec.tex (hunk 1 above)
[the remark this answers is not in the document]
Comment (dnadales) [reply byron/chain/formal-spec/blockchain-spec.tex]: `s_last` is the last seen slot as defined in the state of the `CHAIN` TS.

File: byron/chain/formal-spec/blockchain-spec.tex (hunk 3 above)
[the remark this answers is not in the document]
Comment (nc6) [reply byron/chain/formal-spec/blockchain-spec.tex]: Yes, title case is evil!

File: byron/chain/formal-spec/blockchain-spec.tex [hunk 4]
```diff
@@ -1300,9 +1305,200 @@ \section{Blockchain Extension}
...
+  $$
+  where $t_E$ is the maximum slot number appearing in the blocks contained in
+  $E$, $e_h \leteq \fun{h_e}~e~s$ and $s_h \leteq \fun{h_s}~e~s$,
+  $e_{h_{i-1}} \leteq \fun{h_e}~e~s_{i-1}$ and
+  $s_{h_{i-i}} \leteq \fun{h_s}~e~s_{i-1}$, and $\fun{h_e}$ and $\fun{h_s}$ are
+  the same functions mentioned in Property~\ref{prop:header-only-validation}.
+\end{property}
+
+Property~\ref{prop:roll-back-funk} expresses the fact the there is a function
+that allow us to recover the header-only state by rolling back at most $k$
```
Comment (nc6) [inline byron/chain/formal-spec/blockchain-spec.tex]:
```suggestion
that allows us to recover the header-only state by rolling back at most $k$
```

File: byron/chain/formal-spec/blockchain-spec.tex (hunk 3 above)
[the remark this answers is not in the document]
Comment (dnadales) [reply byron/chain/formal-spec/blockchain-spec.tex]: Yeah, makes sense. Otherwise we always have to think which words to capitalize.

**Pull request merged**